feat: Event Databases back-ups

Molrn opened the pull request.

Molrn committed:
feat: Event Databases back-ups

pascalaubry commented on src/common/papi_web_config.py:
```diff
@@ -396,6 +396,12 @@ def project(self) -> str:
     """ The extension of archives event databases. """
     event_archive_ext: str = 'arch'
 
+    """ The base path where event database back-ups are stored. """
```
For me 'backup' stands for « sauvegarde », so the folder name should simply be 'backup'.

pascalaubry commented on src/data/loader.py:
```diff
@@ -262,3 +264,80 @@ def archives_sorted_by_date(self) -> list[Archive]:
+
+
+@dataclass
+class BackUp:
```
Same as previous comment: this class should be renamed to Backup, or maybe EventBackup.

pascalaubry commented on src/database/sqlite/event_database.py:
```diff
@@ -1288,6 +1301,7 @@ def upgrade(self):
                 f'Your Papi-web version ({papi_web_version}) can not open database {self.file.name} (version '
                 f'{self.version}), please upgrade.'
             )
+        self.create_backup()
```
I has understood that the backup feature was reserved to developers (since upgrading **must** work for any production database), so the backup creation should only done if DEVEL_ENV.

pascalaubry replied:
And BTW the utilities of /utils are not available to non-developers.

pascalaubry reviewed (changes requested):
OK for me if backups are reserved to developers.

Molrn committed:
fixes

- backup sonly created in devel env
- rename back-up to backup
- fix iterdir on not existing dir

pascalaubry reviewed (approved):
Thanks for the changes.
